core/templates/components/forms/custom-forms-directives/edit-thumbnail-modal.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Edit Thumbnail Modal Component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { EditThumbnailModalComponent } from './edit-thumbnail-modal.component';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { NO_ERRORS_SCHEMA, Pipe, SimpleChanges } from '@angular/core';
+import { importAllAngularServices } from 'tests/unit-test-utils';
+
+@Pipe({name: 'translate'})
+class MockTranslatePipe {
+  transform(value: string): string {
+    return value;
+  }
+}
+
+class MockActiveModal {
+  dismiss(): void {
+    return;
+  }
+
+  close(): void {
+    return;
+  }
+}
+
+class mockImageObject {
+  source = null;
+  onload = null;
+  constructor() {
+    this.onload = () => {
+      return 'Fake onload executed';
+    };
+  }
+  set src(url) {
+    this.onload();
+  }
+}
+
+class mockReaderObject {
+  result = null;
+  onload = null;
+  constructor() {
+    this.onload = () => {
+      return 'Fake onload executed';
+    };
+  }
+  readAsDataURL(file) {
+    this.onload();
+    return 'The file is loaded';
+  }
+}
```

Comment:
empty line between blocks and one above constructor

Reply:
done